src/ui_dialogs.ts

```diff
@@ -119,6 +165,79 @@ export function openEditLinkDialog(column: LinkColumn, $header: d3.Selection<Col
   });
 }
 
+
+// Renderer type change
+export function rendererTypeDialog(column: Column, $header: d3.Selection<Column>) {
+  var rendererType = column.getRendererType();
+  const rendererTypeList = column.getRendererList();
+
+  var popup = makeSortPopup($header, 'Change Visualization </br>', rendererTypeList.map(function (d, i) {
+    return `<input type="radio" name="renderertype" value=${d.type}  ${(rendererType === d.type) ? 'checked' : ''}> ${d.label}<br>`;
+
+  }).join('\n'));
+
+
+  function thiselement() {
+    return this === (<any>d3.event).target;
+
+  }
+
+  var rendererContent = d3.selectAll('input[name="renderertype"]');
+  rendererContent.on('change', function () {
+    var that = this;
+    rendererType = that.value;
+    column.setRendererType(that.value);
+
+  });
+
+  // To detect if the mouse click event is triggered outside the sort dialog
+  d3.select('body').on('click', function () {
+    var outside = rendererContent.filter(thiselement).empty();
```

Comment:
I know what the intentions are but not how the line `rendererContent.filter(thiselement).empty();` answers your problem and what the trick behind it is.